src/pages/Embeds/Plio.vue

```diff
@@ -659,7 +686,47 @@ export default {
         if (maximizeButton != undefined) this.mountOnFullscreenPlyr(maximizeButton);
       });
     },
+    checkMovingToTimestampAllowed(timestamp) {
+      if (!this.isSkipEnabled && this.lastAnsweredInteractionIndex < this.numItems - 1) {
+        const firstUnansweredInteraction = this.items[
+          this.lastAnsweredInteractionIndex + 1
+        ];
+        if (timestamp >= firstUnansweredInteraction.time)
+          return firstUnansweredInteraction;
+      }
+    },
+    moveToFirstUnansweredItemTimestampOrPass() {
+      if (this.isSkipEnabled) return false;
+      let timeToInspect = this.player.currentTime;
+
+      if (
+        this.isAnyItemActive &&
+        this.player.currentTime < this.currentItem.time &&
+        this.player.currentTime >= this.currentItem.time - POP_UP_CHECKING_FREQUENCY
+      )
+        timeToInspect += POP_UP_CHECKING_FREQUENCY;
+
+      const firstUnansweredInteraction = this.checkMovingToTimestampAllowed(
+        timeToInspect
+      );
+      if (firstUnansweredInteraction != null) {
+        this.setPlayerTime(firstUnansweredInteraction.time - POP_UP_CHECKING_FREQUENCY);
+        setTimeout(() => {
+          this.toast.error(this.$t("toast.player.cannot_skip_interaction"), {
+            id: "cannotSkipInteraction",
+            position: "bottom-center",
+          });
+        }, 500);
```

Comment:
The settimeout here means that after the player time is set above, 500ms after it, the toast would pop up?
This works fine mostly but on a slow phone (shivani's office phone) what happens is that the toast pops up before.

When you do `this.setPlayerTime(`, plyr tries to buffer the video from that point. Until some amount has been buffered, the question modal that should pop up, doesn't pops up. 

Wouldn't it be a better solution to show this toast when an item has popped up instead of relying on 500ms?

Reply:
haan the trick is to decide if the toast is to be shown in a different function than where this check was done - the function that pops up the item - thik hain - will have a look!